vulnerabilities/importers/alpine_linux.py

```diff
@@ -41,18 +42,26 @@ def advisory_data(self) -> Iterable[AdvisoryData]:
         page_response_content = fetch_response(BASE_URL).content
         advisory_directory_links = fetch_advisory_directory_links(page_response_content)
         advisory_links = []
-        for advisory_directory_link in advisory_directory_links:
-            advisory_directory_page = fetch_response(advisory_directory_link).content
-            advisory_links.extend(
-                fetch_advisory_links(advisory_directory_page, advisory_directory_link)
-            )
-        for link in advisory_links:
-            record = fetch_response(link).json()
-            if not record["packages"]:
-                LOGGER.error(f'"packages" not found in {link!r}')
-                continue
-            yield from process_record(record=record, url=link)
-
+        progress_for_package_fetch = ChargingBar("\tFetching Packages", max=len(advisory_directory_links))
+
+        try:
+            progress_for_package_fetch.start()
+            for advisory_directory_link in advisory_directory_links:
+                advisory_directory_page = fetch_response(advisory_directory_link).content
+                advisory_links.extend(
+                    fetch_advisory_links(advisory_directory_page, advisory_directory_link)
+                )
+            for link in advisory_links:
+                try:
+                    record = fetch_response(link).json()
+                    if not record["packages"]:
+                        LOGGER.error(f'"packages" not found in {link!r}')
+                        continue
+                    yield from process_record(record=record, url=link)
+                finally:
+                    progress_for_package_fetch.next()
+        finally:
+            progress_for_package_fetch.finish()
```

Comment:
The bar is looking good but I don't think using `try finally` is a good idea 
the exception can be discarded if we use `finally` this case .

https://docs.python.org/3/reference/compound_stmts.html#finally-clause

Reply:
Can you please send me the execption I am not able to reproduce.  
I am using finally because i needed to update the progress bar after yielding from generator.

Reply:
I mean theoretically this may cause a problem in the future 
ex: 
```
>>> def f():
...     try:
...             assert 1 == 5
...     finally:
...             return 42
... 
>> f()
42
```

Reply:
Ok I am updating this.
Do you expect it in all importers ?

Reply:
@ziadhany  I am replicating this change to all importers this looks like a good readability fix too

Reply:
Ok , and please don't forget to run these commands to fix the code style and test
```
make valid 
make test  
```

Reply:
Please check @ziadhany 
All testcases passed

Reply:
All requisite changes made
@ziadhany and @TG1999